src/meta/src/model_v2/trx.rs

```diff
@@ -0,0 +1,717 @@
+// Copyright 2023 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::btree_map::{Entry, VacantEntry};
+use std::collections::BTreeMap;
+use std::fmt::Debug;
+use std::ops::{Deref, DerefMut};
+
+use async_trait::async_trait;
+
+use crate::model::MetadataModelResult;
+
+pub type Transaction = sea_orm::DatabaseTransaction;
+
+#[async_trait]
+pub trait Transactional {
+    async fn upsert_in_transaction(&self, trx: &Transaction) -> MetadataModelResult<()>;
+    async fn delete_in_transaction(&self, trx: &Transaction) -> MetadataModelResult<()>;
+}
+
+/// Trait that wraps a local memory value and applies the change to the local memory value on
+/// `commit` or leaves the local memory value untouched on `abort`.
+#[async_trait]
+pub trait ValTransaction: Sized {
```

Comment:
Seems that the only difference between the current on and the original one is the type of the `Transaction`? If so, shall try introducing a associated type `Transaction` or a generic parameter in the trait definition so that we may avoid completely duplicating the original code.

Reply:
Yes the only several difference including `Transaction` and sync/async method.
The duplicated code is because model V1 and model V2 may coexist for some time. In the end I'll remove this duplicate file.